src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -51,33 +50,40 @@ private BoundExpression BindIsPatternExpression(IsPatternExpressionSyntax node,
                         (DeclarationPatternSyntax)node, operandType, hasErrors, diagnostics);
 
                 case SyntaxKind.ConstantPattern:
+                    var constantPattern = (ConstantPatternSyntax)node;
                     return BindConstantPattern(
-                        (ConstantPatternSyntax)node, operandType, hasErrors, diagnostics, wasSwitchCase);
+                        constantPattern, operandType, constantPattern.Expression, hasErrors, diagnostics, out bool wasExpression);
 
                 default:
                     throw ExceptionUtilities.UnexpectedValue(node.Kind());
             }
         }
 
-        private BoundConstantPattern BindConstantPattern(
-            ConstantPatternSyntax node,
+        internal BoundConstantPattern BindConstantPattern(
+            CSharpSyntaxNode node,
             TypeSymbol operandType,
+            ExpressionSyntax patternExpression,
             bool hasErrors,
             DiagnosticBag diagnostics,
-            bool wasSwitchCase)
+            out bool wasExpression)
         {
-            bool wasExpression;
-            return BindConstantPattern(node, operandType, node.Expression, hasErrors, diagnostics, out wasExpression, wasSwitchCase);
+            var innerExpression = patternExpression.SkipParens();
+            if (innerExpression.Kind() == SyntaxKind.DefaultLiteralExpression)
```

Comment:
Any chance `innerExpression` could be `null` here? (`case ():` maybe?)

Reply:
No. That is a positional pattern, but even if parsed as a pattern the parser will insert a missing identifier.